packages/frint-router-react/src/Route.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getMountableComponent } from 'frint-react';
+/* eslint-enable import/no-extraneous-dependencies */
+
+export default class Route extends React.Component {
+  static contextTypes = {
+    app: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    path: PropTypes.string,
+    exact: PropTypes.bool,
+    computedMatch: PropTypes.object,
+    component: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+    app: PropTypes.func,
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this._routerSubscription = null;
+    this._appInstance = null;
+
+    this.state = {
+      component: null,
+      matched: null,
+    };
+  }
+
+  componentWillMount() {
+    this._calculateMatchedState(this.props);
+    this._calculateComponentState(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this._calculateMatchedState(nextProps);
+    this._calculateComponentState(nextProps);
+  }
+
+  _calculateMatchedState(nextProps) {
+    if (nextProps.computedMatch) {
+      // in case it was subscribed before
+      this._unsubscribeFromRouter();
+    } else if (nextProps.path) {
+      if (!this._routerSubscription || (nextProps.path !== this.props.path) || (nextProps.exact !== this.props.exact)) {
+        this._unsubscribeFromRouter();
+
+        this._routerSubscription = this.context.app
+          .get('router')
+          .getMatch$(nextProps.path, {
+            exact: nextProps.exact,
+          })
+          .subscribe((matched) => {
+            this.setState({
+              matched,
+            });
+          });
+      }
+    }
+  }
+
+  _calculateComponentState(nextProps) {
+    if (nextProps.component) {
+      // component
+      this._destroyRouteApp();
+
+      this.setState({
+        component: nextProps.component,
+      });
+    } else if (nextProps.app && (this._appInstance === null || nextProps.app !== this.props.app)) {
+      // app
+      this._destroyRouteApp();
+
+      const RouteApp = nextProps.app;
+
+      this._appInstance = new RouteApp({
```

Comment:
we can't `get` because the instance does not live anywhere. these are App classes that are passed to `<Route app={App} />` components.

since they are not registered to the Root app, like how we do it for Apps targeting Regions, they can't be obtained later from somewhere else. these Apps' lifecycle are contained by the `<Route>`.